### ISSUE TITLE
this library's resetGenericPassword method is not working

### ISSUE THREAD

**Issue by ceyanesb** [oblador/react-native-keychain#590]
I am developing an app and I want to clean my credentials when the app is installed for and opened for the first time or when the user logs out.

Every time I login, then uninstall and then install the app again i get the credentials which I shouldn't get.

**Comment by BraveEvidence** [comment]
This will help https://www.youtube.com/watch?v=J0OSn7s9YiA&list=PLQhQEGkwKZUrempLnmxjt7ZCZJu1W3p2i&index=15

**Comment by raza-jamil-reckon** [comment]
Seeing the same thing, our logout stopped working all of a sudden and I've tracked it back to `resetGenericPassword` not doing anything. It returns a true value, which means it thinks it's succeeded but it doesn't actually reset the keychain.

**Comment by jde8aa** [comment]
In my app (RN 0.71.12) when using react-native-keychain v8.1.1, if I call the **resetGenericPassword()** method after having saved my username and password previously, and then call **getGenericPassword()** afterwards it will return garbage data where the username is null and the password is some random string of characters instead of "false" like it should.

**Comment by JuozasIzi1** [comment]
@DorianMazur this is behaviour also happens for me also is this planned to look into? this really bad bug, that could leak users data to another and have GDPR issues.

**Comment by DorianMazur** [comment]
Hey @JuozasIzi1  Yes, I will be working on it. I'm currently working on two PRs: [one](https://github.com/oblador/react-native-keychain/pull/646) adds end-to-end tests, which should help cover important scenarios. I'll try to recreate and fix this bug.

**Comment by ezescigo** [comment]
Same happening to me on RN v0.71.14 and react-native-Keychain v9.0.0.

**Comment by DorianMazur** [comment]
I don't think this is happening in 9.0.0 or 9.1.0. Can you provide example repository or reproduction steps?

**Comment by ezescigo** [comment]
unfortunately I can't, it's a private repository. But I'm not able to delete the saved credentials, so yeah, there's an issue with it at least on my case. Sadly I will need to use other library.

**Comment by ryd0rz** [comment]
This still isn't working for me on version 9.2.2. Are there any workarounds?
